codegen/src/main/ops/org/nd4j/codegen/ops/Linalg.kt

```diff
@@ -206,6 +208,34 @@ fun Linalg() =  Namespace("Linalg") {
             """.trimIndent()
         }
     }
+
+    Op("tri") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "Tri"
+
+        Arg(DataType.INT, "params") {count = AtLeast(3);"shapes of tensor and sub-diagonal number at and below which the array is filled"; }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             An array with ones at and below the given diagonal and zeros elsewhere.
+            """.trimIndent()
+        }
+    }
+
+    Op("triu") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "Triu"
+
+        Input(DataType.NUMERIC, "input") {"Input tensor"}
+        Arg(DataType.INT, "diag") {"diagonal"; defaultValue = 0}
+        Output(DataType.FLOATING_POINT, "output")
```

Comment:
Let's also add an arg that allows the user to specify the output datatype (default value: float)